### PR TITLE
TENT-1556 - support up/down HTML tooltips

### PR DESCRIPTION
**Pull request by rossedfort**
Support HTML tooltips above and below triggers. [TENT-1556](https://jira.sendgrid.net/browse/TENT-1556)

- [x] Includes link to associated Github Issue or JIRA ticket
- [x] Has descriptive PR title and body
- [x] All changes are backwards compatible
- [x] Code reviewed by a SendGrid frontend lead

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -141,7 +141,7 @@ More information: https://semver.org/
 - `npm run ci-test`: Test used for the CI build (doesn't use interactive mode).
 - `npm run build-storybook`: Build a deployable version of the Storybook.
 - `npm run prepare`: Used by npm to build the assets before publishing.
-- `npm run run image-snapshots`: Builds a static index.html file and runs image snapshot tests.
+- `npm run image-snapshots`: Builds a static index.html file and runs image snapshot tests.
```
Comment (Ditofry) [inline README.md]: woah... and this wasn't causing problems?

File: src/tooltip.tsx [hunk 2]
```diff
@@ -177,6 +167,57 @@ export class HTMLTooltip extends React.Component<
...
+
+  /**
+   * determines vertical & horizontal positioning for the tooltip content.
+   * NOTE: this does NOT determine positioning for the ::after pseudoelement, i.e. the caret/arrow of the tooltip
+   *
+   * 'tooltip-js-content' always exists in the DOM for each tooltip. It's displayed/hidden via CSS opacity.
+   * This is because we want the tooltip content to animate in when a user triggers it.
+   * We move the tooltip content well off the screen, because we don't want it to block users' interactions
+   * with other elements on the page. If we didn't do this,
+   * a user could trigger a tooltip by hovering over its invisible content.
```
Comment (Ditofry) [inline src/tooltip.tsx]: Was this a bug up until this PR?

File: README.md (hunk 1 above)
Comment (rossedfort) [reply README.md]: 🤷‍♂️ I think it was an artifact of the `yarn` -> `npm` migration

File: src/tooltip.tsx (hunk 2 above)
Comment (rossedfort) [reply src/tooltip.tsx]: Nope, this particular style used to be applied via css, so nothing new technically, but this PR moves all positioning related styles into the JS for (I hope) clarity. I just left the comment to make note of why we were positioning the tooltip -10,000px left before it was displayed.

File: README.md (hunk 1 above)
Comment (rossedfort) [reply README.md]: (so a fairly recent addition, likely no one has run into it yet)

**Comment by thinkingserious** [comment]
Hello @rossedfort,<br /><br/>Thanks again for the PR!<br />                                              <br />We appreciate your contribution and look forward to                                              continued collaboration. Thanks!                                             <br /><br />Team SendGrid DX

**Pull request merged**